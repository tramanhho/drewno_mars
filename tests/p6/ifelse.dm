main : () void {
    if (true) {
        give "hi";
    } else {
        give "bye";
    }
}

main2 : () int {
	a : bool = true;
	// a = 24Kmagic;
    if (a) {
        return 1;
    } else {
        return 0;
    }
}