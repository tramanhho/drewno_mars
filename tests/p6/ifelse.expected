[BEGIN GLOBALS]

[END GLOBALS]

[BEGIN main LOCALS]
a-bool
[END main LOCALS]

main:    enter main
        [a] := 24Kmagic
        IFZ [a] goto lbl_2
        WRITE 1
        goto lbl_1

lbl_2:    nop
        WRITE 0

lbl_1:    nop
        setret 0
        goto lbl_0
lbl_0:    leave main